### PR TITLE
[#42][Chore] Replace cross_file with flutter_dotenv

### LINKED ISSUE #42: [Chore] Replace cross_file with flutter_dotenv
## Why

Replace `cross_file` with `flutter_dotenv` to keep environment secrets. 

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by Wadeewee**
- Close #42

## What happened 👀

Replace [cross_file](https://pub.dev/packages/cross_file) with [flutter_dotenv](https://pub.dev/packages/flutter_dotenv#usage) to keep environment secrets. 

## Insight 📝

To use `flutter_dotenv` it recommends defining `.env` in `pubspec.yaml` under the `assets` directory 👇 

Screen Shot 2566-07-17 at 15 29 39

By following this step above ☝️ We need the scripts to setup `ENV` for `CI/CD` pipelines If not we will face this error 👇 

Screen Shot 2566-07-17 at 15 32 15

## Proof Of Work 📹

- The app can run normally

Screen Shot 2566-07-17 at 15 24 55

### CODE REVIEW

File: macos/Runner/DebugProfile.entitlements [hunk 1]
```diff
@@ -8,5 +8,7 @@
 	<true/>
 	<key>com.apple.security.network.server</key>
 	<true/>
+	<key>com.apple.security.network.client</key>
+	<true/>
```
Comment (Wadeewee) [inline macos/Runner/DebugProfile.entitlements:12]: Please ignore this changed 🙏

File: macos/.gitignore [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Flutter-related
 **/Flutter/ephemeral/
 **/Pods/
+**/Flutter/GeneratedPluginRegistrant.*
+**/Runner.*
```
Comment (luongvo) [inline macos/.gitignore:5]: Ignore `**/Flutter/GeneratedPluginRegistrant.*` makes sense. But why do we ignore `**/Runner.*`?

image

Comment (Wadeewee) [reply macos/.gitignore:5]: @luongvo I will auto-generate when execute running `macOs` So, I think we don't need push them up to the repo 🤔

Comment (luongvo) [reply macos/.gitignore:5]: @Wadeewee I don't think excluding them is a best practice in iOS development. Especially when we have to set up multiple app flavors (scheme in iOS) 🤔 

image

Comment (Wadeewee) [reply macos/.gitignore:5]: Ah, I see @luongvo I will remove it 🚀

**Review by kaungkhantsoe — APPROVED** [review]

**Review by ducbm051291 — APPROVED** [review]

**Pull request merged**